xdsl/passes.py

```diff
@@ -151,23 +159,30 @@ def pipeline_pass_spec(self, *, include_default: bool = False) -> PipelinePassSp
         return PipelinePassSpec(self.name, args)
 
 
-# Git Issue: https://github.com/xdslproject/xdsl/issues/1845
-def get_pass_argument_names_and_types(arg: type[ModulePassT]) -> str:
+class PassArgOption(NamedTuple):
+    """ """
+
+    name: str
+    type_str: str
+    default_value_str: str | None = None
```

Comment:
```suggestion
    value_type: str
    default_value: str | None = None
```
I know these are for object reconstruction, but `_str` doesn't add much info?

Reply:
You're right. I was initially hoping to unstringify it all the way, and pass structured information around, but it seemed like too big a change. I think it makes sense to give things nice names even if we might change the types later.